src/osx/core/display.cpp

```diff
@@ -196,31 +228,44 @@ int wxDisplayFactoryMacOSX::GetFromPoint(const wxPoint& p)
     CGDisplayErr err = CGGetDisplaysWithPoint(thePoint, 1, &theID, &theCount);
     wxASSERT(err == CGDisplayNoErr);
 
-    int nWhich = wxNOT_FOUND;
-
     if (theCount)
-    {
-        theCount = GetCount();
-        CGDirectDisplayID* theIDs = new CGDirectDisplayID[theCount];
-        err = wxOSXGetDisplayList(theCount, theIDs, &theCount);
-        wxASSERT(err == CGDisplayNoErr);
+        return wxOSXGetDisplayFromID(theID);
 
-        for (nWhich = 0; nWhich < (int) theCount; ++nWhich)
-        {
-            if (theIDs[nWhich] == theID)
-                break;
-        }
+    return wxNOT_FOUND;
+}
 
-        delete [] theIDs;
+int wxDisplayFactoryMacOSX::GetFromWindow(const wxWindow *window)
+{
+    wxCHECK_MSG( window, wxNOT_FOUND, "window can't be NULL" );
 
-        if (nWhich == (int) theCount)
+    wxNonOwnedWindow* tlw = window->MacGetTopLevelWindow();
+    int x,y,w,h;
+
+    tlw->GetPosition(&x, &y);
+    tlw->GetSize(&w, &h);
+
+    CGRect r = CGRectMake(x, y, w, h);
+    CGDisplayCount theCount;
+    CGDisplayErr err = CGGetDisplaysWithRect(r, 0, NULL, &theCount);
+    wxASSERT(err == CGDisplayNoErr);
+
+    wxScopedArray<CGDirectDisplayID> theIDs(theCount);
+    err = CGGetDisplaysWithRect(r, theCount, theIDs.get(), &theCount);
+    wxASSERT(err == CGDisplayNoErr);
+
+    for ( int i = 0 ; i < theCount; ++i )
+    {
+        // find a screen intersecting having the same contentScale as the window itself
+        wxCFRef<CGDisplayModeRef> mode = CGDisplayCopyDisplayMode(theIDs[i]);
+        size_t width = CGDisplayModeGetWidth(mode);
+        size_t pixelsw = CGDisplayModeGetPixelWidth(mode);
```

Comment:
It would be nice to reuse the same code in `wxDisplay::GetScaleFactor()` added in the previous commit and here, e.g. by adding a common helper function.

Reply:
yes, I'll do that